Meadowlark-js/src/repository/ElasticsearchRepository.ts

```diff
@@ -96,63 +102,33 @@ async function performSqlQuery(client: Client, query: string): Promise<any> {
   });
 }
 
-/**
- * For Education Organization and Student security, combine WHERE clauses
- * if necessary. Assumes at least one is not empty.
- */
-function fullDirectWare(edOrgDirectWhere: string, studentDirectWhere: string): string {
-  if (edOrgDirectWhere === '') return studentDirectWhere;
-  if (studentDirectWhere === '') return edOrgDirectWhere;
-  return `(${edOrgDirectWhere} OR ${studentDirectWhere})`;
-}
-
 /**
  * Entry point for querying with Elasticsearch, given entity info, the original http
  * request query parameters, and a Security object for possible filtering
  */
 export async function queryEntityList(
   entityInfo: EntityInfo,
   queryStringParameters: object,
-  security: Security,
+  paginationParameters: PaginationParameters,
   awsRequestId: string,
 ): Promise<GetResult> {
-  Logger.debug(`ElasticsearchRepository.queryEntityList security info: ${JSON.stringify(security)}`, awsRequestId);
-
   const client = await getElasticsearchClient(awsRequestId);
 
-  let directResults: any = {};
-  const whereConditionsFromQueryString = whereConditionsFrom(queryStringParameters);
   Logger.debug(`Building query`, awsRequestId);
+
+  let results: any = {};
   try {
-    // direct relationship
-    let query = `SELECT info FROM ${indexFromEntityInfo(entityInfo)} WHERE `;
-
-    // no security if no claims (for demo/testing reasons)
-    if (security.edOrgIds.length === 0 && security.studentIds.length === 0) {
-      query += whereConditionsFromQueryString;
-    } else {
-      let edOrgDirectWhere = '';
-      let studentDirectWhere = '';
-
-      // ed org direct security
-      if (security.edOrgIds.length > 0) {
-        edOrgDirectWhere = `extractedEdOrgId IN (${security.edOrgIds.map((edOrgId) => `'${edOrgId}'`).join(', ')})`;
-      }
-      // student direct security
-      if (security.studentIds.length > 0) {
-        studentDirectWhere += `extractedStudentId IN (${security.studentIds.map((edOrgId) => `'${edOrgId}'`).join(', ')})`;
-      }
-
-      const directWhere = fullDirectWare(edOrgDirectWhere, studentDirectWhere);
-      query += `${whereConditionsFromQueryString} AND ${directWhere}`;
-    }
+    let query = `SELECT info FROM ${indexFromEntityInfo(entityInfo)}`;
+    if (Object.entries(queryStringParameters).length > 0) query += ` WHERE ${whereConditionsFrom(queryStringParameters)}`;
+    if (paginationParameters.limit != null) query += ` LIMIT ${paginationParameters.limit}`;
+    if (paginationParameters.offset != null) query += ` OFFSET ${paginationParameters.offset}`;
 
     Logger.debug(`ElasticsearchRepository.queryEntityList executing query: ${query}`, awsRequestId);
```

Comment:
(random location for this comment, nothing to do with line 126)

I'm a bit concerned [about this](https://opensearch.org/docs/latest/opensearch/ux/#paginate-results):

> "Querying for pages deep in your results can have a significant performance impact, so OpenSearch limits this approach to 10,000 results."

I think this means that larger pagination, like "get all students" when > 10,000 students, might need to use the `search_after` functionality. So I'll create a ticket for more research. We're in good enough shape with this release.

Reply:
👍

Reply:
https://tracker.ed-fi.org/browse/RND-204